RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackProducer.cc

```diff
@@ -15,38 +15,60 @@
 
-#include "PixelTrackProducer.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "RecoPixelVertexing/PixelTrackFitting/interface/PixelTrackReconstruction.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
 #include "storeTracks.h"
 
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+namespace edm {
+  class Event;
+  class EventSetup;
+  class ParameterSet;
+  class ConfigurationDescriptions;
+}  // namespace edm
+class TrackerTopology;
+
 using namespace pixeltrackfitting;
 using edm::ParameterSet;
 
-PixelTrackProducer::PixelTrackProducer(const ParameterSet& cfg) : theReconstruction(cfg, consumesCollector()) {
-  edm::LogInfo("PixelTrackProducer") << " construction...";
-  produces<reco::TrackCollection>();
-  produces<TrackingRecHitCollection>();
-  produces<reco::TrackExtraCollection>();
-}
+class PixelTrackProducer : public edm::stream::EDProducer<> {
+public:
+  explicit PixelTrackProducer(const edm::ParameterSet& cfg)
+      : theReconstruction(cfg, consumesCollector()), htTopoToken_(esConsumes()) {
+    edm::LogInfo("PixelTrackProducer") << " construction...";
+    produces<reco::TrackCollection>();
+    produces<TrackingRecHitCollection>();
+    produces<reco::TrackExtraCollection>();
+  }
+
+  ~PixelTrackProducer() override {}
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
 
-PixelTrackProducer::~PixelTrackProducer() {}
+    desc.add<std::string>("passLabel", "pixelTracks");  // What is this? It is not used anywhere in this code.
+    PixelTrackReconstruction::fillDescriptions(desc);
 
-void PixelTrackProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
-  edm::ParameterSetDescription desc;
+    descriptions.add("pixelTracks", desc);
+  }
 
-  desc.add<std::string>("passLabel", "pixelTracks");  // What is this? It is not used anywhere in this code.
-  PixelTrackReconstruction::fillDescriptions(desc);
+  void produce(edm::Event& ev, const edm::EventSetup& es) override {
+    LogDebug("PixelTrackProducer, produce") << "event# :" << ev.id();
 
-  descriptions.add("pixelTracks", desc);
-}
+    TracksWithTTRHs tracks;
+    theReconstruction.run(tracks, ev, es);
+    auto htTopo = es.getData(htTopoToken_);
 
-void PixelTrackProducer::produce(edm::Event& ev, const edm::EventSetup& es) {
-  LogDebug("PixelTrackProducer, produce") << "event# :" << ev.id();
+    // store tracks
+    storeTracks(ev, tracks, htTopo);
+  }
 
-  TracksWithTTRHs tracks;
-  theReconstruction.run(tracks, ev, es);
-  edm::ESHandle<TrackerTopology> httopo;
-  es.get<TrackerTopologyRcd>().get(httopo);
+private:
+  PixelTrackReconstruction theReconstruction;
+  const edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> htTopoToken_;
+};
 
-  // store tracks
-  storeTracks(ev, tracks, *httopo);
-}
+DEFINE_FWK_MODULE(PixelTrackProducer);
```

Comment:
I already said I was takling care of this in my PR that is coming soon

Reply:
I can keep this PR on hold and apply on top of yours when it is in.

Reply:
I think that this PR should be closed. None of the modification in here is neither essential nor an improvement.